### DISCUSSION TITLE
Script in bottlerocket user-data

### DISCUSSION THREAD

**Discussion by ghost** [bottlerocket-os/bottlerocket#1343] [Q&A] [answered]
Hi,
I am using bottlerocket AMI in AWS. The nodes are successfully joinig the cluster, now I have to place some custom node label, that will contain availability zone nome in it. Example:
nodeGroup-bottlerocket-us-east-1a.
With Amazon LInux AMI i run the curl request to metadata:
ENI_CONFIG=${NODE_TYPE}-$(curl 169.254.169.254/latest/meta-data/placement/availability-zone)

```
/etc/eks/bootstrap.sh \
    --b64-cluster-ca '${B64_CLUSTER_CA}' \
    --apiserver-endpoint '${API_SERVER_URL}' ${bootstrap_extra_args} \
    --kubelet-extra-args "--node-labels=node.eniconfig/configname=$${ENI_CONFIG}" '${CLUSTER_NAME}'
```
is there any way to run script in bottlerocket and put the AZ-specific label?

**Comment by bcressey** [comment, marked as answer]
Today there's no way to do exactly what you're asking for, outside of a custom build.

We do have the ability to populate settings based on a template at runtime; that's how [host containers](https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/models/shared-defaults/aws-host-containers.toml) are configured to pull from the regional ECR repository. However, there's not currently a way to define new templates or override them via user-data.

We're also in the early stages of designing a "bootstrap container" interface that would let you run a script inside a container during first boot, so you could do something like this:
```
INSTANCE_TYPE="$(curl -s http://169.254.169.254/latest/meta-data/instance-type)"
AZ="$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)"
apiclient set --json '{"kubernetes":{"node-labels":{"node.eniconfig/configname":"'${INSTANCE_TYPE}-${AZ}'"}}}'
```

For your use case, could you work with a combination of two labels instead? `node.kubernetes.io/instance-type` and `topology.kubernetes.io/zone` are set by default and have the data you're after, though not in the same label.

**Reply by bcressey** [reply]
Bootstrap containers are now available as of the 1.0.8 release, so you'd be able to use a script like the one above to add a dynamically generated label to the settings.
